api/src/main/java/org/openmrs/module/reporting/cohort/definition/library/BuiltInCohortDefinitionLibrary.java

```diff
@@ -171,4 +172,12 @@ public CohortDefinition getPatientsWithEnrollment(){
             "locationList", "locations"
         );
     }
+
+    @DocumentedDefinition("encounterLocationSearch")
```

Comment:
Okay, @mseaton. I will add the timeQualifier as an optional property and rename the method to encounterSearchAdvanced as you pointed.
I will also update the JIRA ticket so it better describes these changes.